net-p2p/mldonkey/mldonkey-3.1.7.ebuild

```diff
@@ -0,0 +1,187 @@
+
+IUSE="bittorrent doc fasttrack gd gnutella gtk guionly magic +ocamlopt upnp"
+
+DESCRIPTION="Multi-network P2P application written in Ocaml, with Gtk, web & telnet interface"
+HOMEPAGE="http://mldonkey.sourceforge.net/"
+SRC_URI="https://github.com/ygrek/mldonkey/releases/download/release-${PV//./-}-2/${P}-2.tar.bz2"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~arm ~hppa ~ia64 ~ppc ~ppc64 ~sparc ~x86"
+
+RDEPEND="dev-lang/perl
+	dev-ml/camlp4:=
+	guionly? ( >=gnome-base/librsvg-2.4.0
+			>=dev-ml/lablgtk-2.6 )
+	gtk? ( >=gnome-base/librsvg-2.4.0
+			>=dev-ml/lablgtk-2.6[svg] )
+	gd? ( >=media-libs/gd-2.0.28[truetype] )
+	magic? ( sys-apps/file )
+	upnp? ( net-libs/miniupnpc:= net-libs/libnatpmp:= )
+"
+
+DEPEND="${RDEPEND}
+	acct-user/p2p
+	<dev-lang/ocaml-4.10[ocamlopt?]
+	bittorrent? (
+		|| (
+			<dev-lang/ocaml-4.06
+			(
+				>=dev-lang/ocaml-4.06
+				dev-ml/num
+			)
+		)
+	)"
+
+RESTRICT="!ocamlopt? ( strip )"
+
+MLUSER="p2p"
+
+S="${WORKDIR}/${P}-2"
+
+pkg_setup() {
+	if use gtk; then
+		echo ""
+		einfo "If the compile with gui fails, and you have updated Ocaml"
+		einfo "recently, you may have forgotten that you need to run"
+		einfo "/usr/portage/dev-lang/ocaml/files/ocaml-rebuild.sh"
+		einfo "to learn which ebuilds you need to recompile"
+		einfo "each time you update Ocaml to a different version"
+		einfo "see the Ocaml ebuild for details"
+		echo ""
+	fi
+
+	# dev-lang/ocaml creates its own objects but calls gcc for linking, which will
+	# results in relocations if gcc wants to create a PIE executable
+	if gcc-specs-pie ; then
+		append-ldflags -nopie
+		ewarn "Ocaml generates its own native asm, you're using a PIE compiler"
+		ewarn "We have appended -nopie to ocaml build options"
+		ewarn "because linking an executable with pie while the objects are not pic will not work"
+	fi
+}
+
+src_prepare() {
+	cd "${S}"/config
+	eautoconf
+	cd "${S}"
+	use ocamlopt || sed -i -e "s/ocamlopt/idontwantocamlopt/g" "${S}/config/configure" || die "failed to disable ocamlopt"
+
+	default
+}
+
+src_configure() {
+	# the dirs are not (yet) used, but it doesn't hurt to specify them anyway
+
+	# onlygui	Disable all nets support, build only chosen GUI
+
+	if use gtk || use guionly; then
+		myconf="--enable-gui=newgui2"
+	else
+		myconf="--disable-gui"
+	fi
+
+	if use guionly; then
+		myconf="${myconf} --disable-multinet --disable-donkey"
+	fi
+
+	cd "${S}"
+
+	local my_extra_libs
+	if use gd; then
+		my_extra_libs="-lpng"
+	fi
+
+	econf LIBS="${my_extra_libs}"\
+		--sysconfdir=/etc/mldonkey \
+		--sharedstatedir=/var/mldonkey \
+		--localstatedir=/var/mldonkey \
+		--enable-checks \
+		--disable-batch \
+		$(use_enable bittorrent) \
+		$(use_enable fasttrack) \
+		$(use_enable gnutella) \
+		$(use_enable gnutella gnutella2) \
+		$(use_enable gd) \
+		$(use_enable magic) \
+		$(use_enable upnp upnp-natpmp) \
+		--disable-force-upnp-natpmp \
+		${myconf}
+}
+
+src_compile() {
+	export OCAMLRUNPARAM="l=256M"
+	emake -j1 # Upstream bug #48
```

Comment:
Is this still true?

Reply:
Yes, it is. Actually, that workaround was not present in the ebuild so far and I added it for this version because compilation was failing.

I also reported it upstream: https://github.com/ygrek/mldonkey/issues/48

Reply:
Marking as resolved.